### PR TITLE
challenge 1 - 8 curs8 Core Java APIs

### PR DESCRIPTION
**Pull request by MarianP24**
ma ajuta sa imi mai las comentarii prin cod sa pot intelege mai bine linia respectiva si daca ma intorc la ea sa stiu ce am facut :)

### CODE REVIEW

**Comment by ionutbaur** [comment]
> ma ajuta sa imi mai las comentarii prin cod sa pot intelege mai bine linia respectiva si daca ma intorc la ea sa stiu ce am facut :)

Cu siguranta ca o sa iti las acolo unde e cazul. Insa daca vad ca e ok si corect, presupun ca stii deja ce ai facut acolo si ai inteles problema si rezolvarea, nu?

File: Session9/src/com/itschool/homework/Session9Challenge1.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.itschool.homework;
+
+import java.util.Scanner;
+
+public class Session9Challenge1 {
```
Comment (ionutbaur) [inline Session9/src/com/itschool/homework/Session9Challenge1.java:5]: I think these all challenges are from session8, not 9.

File: Session9/src/com/itschool/homework/Session9Challenge3.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+public class Session9Challenge3 {
+    public static void main(String[] args) {
+        Session9Challenge3 run = new Session9Challenge3();
+        run.ChallengeRun3();
+    }
+
+    public void ChallengeRun3() {
+        StringBuilder concatenate = new StringBuilder();
+        List<String> stringList = List.of("How ","are ","you ","today","?");
```
Comment (ionutbaur) [inline Session9/src/com/itschool/homework/Session9Challenge3.java:13]: nice!

File: Session9/src/com/itschool/homework/Session9Challenge4.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public void challengeRun4() {
+        StringBuilder remove = new StringBuilder("Let's go for a ride");
+        System.out.println(remove);
+        for (int i = 0; i < remove.length(); i++) {
+            char ch = remove.charAt(i);
+            if (isVowel(ch)) {
+                remove.deleteCharAt(i);
+                i--;
```
Comment (ionutbaur) [inline Session9/src/com/itschool/homework/Session9Challenge4.java:16]: I don't get it. What is the purpose of this decrement? Removing it has the same outcome.

File: Session9/src/com/itschool/homework/Session9Challenge7.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+public class Session9Challenge7 {
+    public static void main(String[] args) {
+        Session9Challenge7 run = new Session9Challenge7();
+        run.challengeRun7();
+    }
+
+    public void challengeRun7() {
+        StringBuilder dupli = new StringBuilder("Football challenge");
+        System.out.println(dupli);
+        for (int i = 0; i < dupli.length() - 1; i++)
```
Comment (ionutbaur) [inline Session9/src/com/itschool/homework/Session9Challenge7.java:12]: not recommended to have the loop body without `{ }`.

**Review by ionutbaur — APPROVED** [review]

File: Session9/src/com/itschool/homework/Session9Challenge1.java (hunk 1 above)
Comment (MarianP24) [reply Session9/src/com/itschool/homework/Session9Challenge1.java:5]: yes indeed, i was thinking if i should let them inside session9 or 8 but i went straight for the last one, 9. I apologise for the confusion

**Pull request merged**